### PR TITLE
Helix Separation

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

This PR brings is the sister PR to [this HelixToolkit PR](https://github.com/helix-toolkit/helix-toolkit/pull/172). It enacts the separation of the types and effects that are created specifically for Dynamo, from the HelixToolkit code base.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
  - We no longer have Dynamo-specific code in Helix!
- [x] The level of testing this PR includes is appropriate
  - Functionality remains the same. Existing tests will verify that this works.
- [x] User facing strings, if any, are extracted into `*.resx` files
  - No new user-facing strings have been added.

### Reviewers

@ramramps

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 1]
```diff
@@ -360,6 +351,37 @@ private void InitializeHelix()
             }
         }
 
+        void RegisterRenderDynamoEffect(EffectInitializationEventArgs args)
```
Comment (ikeough) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs]: This method is the event handler for a new event added to Helix which allows for third-party registration of `Effect` objects.

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 2]
```diff
@@ -1180,7 +1202,7 @@ private void AggregateRenderPackages(PackageAggregationParams parameters)
                 }
                 else
                 {
-                    meshGeometry3D = new DynamoGeometryModel3D()
+                    meshGeometry3D = new DynamoGeometryModel3D(renderTechnique)
```
Comment (ikeough) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs]: The `DynamoGeometryModel3D` type now takes the `RenderTechnique` as a parameter to the constructor. Previously, in Helix, these render techniques were stored in a static dictionary.

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 3]
```diff
@@ -153,19 +156,6 @@ public Viewport3DX View
         public PhongMaterial SelectedMaterial { get; set; }
 
         public Transform3D Model1Transform { get; private set; }
-        
-        public RenderTechnique RenderTechnique
```
Comment (ikeough) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs]: This property is removed as it is no longer required to bind a `RenderTechnique` to the helix view. Each of the types we render, points, curves, and meshes, sets their own `RenderTechnique`. This render technique for the scene is not required.

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 4]
```diff
@@ -302,6 +290,9 @@ private void SetupScene()
             SetCameraToDefaultOrientation();
 
             DrawGrid();
+
+            renderTechnique = new RenderTechnique("RenderDynamo");
```
Comment (ikeough) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs]: Here we create our custom `RenderTechnique` and we register the event handler to register this effect when Helix is initializing.

File: src/DynamoCoreWpf/DynamoGeometryModel3D.cs [hunk 5]
```diff
@@ -0,0 +1,290 @@
+﻿using System;
```
Comment (ikeough) [inline src/DynamoCoreWpf/DynamoGeometryModel3D.cs]: This type is brought over from the Helix code-base. It's the underlying class for every mesh drawn in Dynamo.

File: src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs [hunk 6]
```diff
@@ -302,6 +290,9 @@ private void SetupScene()
...
+            EffectsManager.Instance.InitializingEffects += RegisterRenderDynamoEffect;
```
Comment (ramramps) [inline src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs]: Do we need to unregister this event?

**Comment by ramramps** [comment]
@ikeough  Just one question, then LGTM,.

**Pull request closed without merging**